### PR TITLE
V1.0

### LINKED ISSUE #13: Library structure
This thread exists as a place to discuss the modules, functions, data structures, and their names included in the library. In version `1.0.0` names must be agreed upon and remain immutable for backward compatibility. Until then, we should experiment as much as possible and find out what works best.

#### gltf

 - The main crate, intended for navigating glTF JSON.
 - Intentionally limited in scope.

#### gltf-json

 - Child crate for `gltf`.
 - Exists primarily to reduce compile times for `gltf`.
 - `semver` matches that of `gltf` for convenience but is otherwise unstable.

 #### gltf-importer

- Reference importer with a simple API.
- May be used by run times to read from the file system.
- Handles base 64 decoding and distinguishing binary glTF on behalf of the user.
- Optionally loads buffers and images (to do).

#### gltf-utils

- Utility functions to complement the `gltf` crate.
- Addresses cases where access to pre-existing buffer / image data is required.
- ~~May remain unstable post `gltf 1.0`.~~

### PR DESCRIPTION
**Pull request by alteous**
Closes #13 and #12

### CODE REVIEW

**Comment by alteous** [comment]
> Ooh - time for 1.0 already?

I think so. I've made this PR mainly for visibility. I don't expect this to be merged within the next week.

**Comment by alteous** [comment]
>  I'd wish the library would make image loading a bit simpler or at least provide a complete example).

How could we address this?  Would a `read_image` function in `gltf-utils` plus `gltf-importer` returning `(Gltf, Buffers, Images)` be suitable?

**Comment by bwasty** [comment]
> >  I'd wish the library would make image loading a bit simpler or at least provide a complete example).
> 
> How could we address this?  Would a `read_image` function in `gltf-utils` plus `gltf-importer` returning `(Gltf, Buffers, Images)` be suitable?

I've also thought about `gltf-utils`, not sure if it has to be in `gltf-importer` as well. Essentially it should encapsulate [this block](https://github.com/three-rs/three/blob/139c22c757d94b75dd8deda90f2934a08b0ed2b3/src/factory/load_gltf.rs#L23-L54) minus the image decoding - give it a `gltf::image::Image`, get raw binary data without caring whether it comes from a file, GLB or Data URI. Not sure in how far a really simple interface is feasible. It doesn't have to be super convenient, but saving most of those lines would be nice.

File: src/buffer.rs [hunk 1]
```diff
@@ -13,6 +13,15 @@ use Gltf;
 
 pub use json::buffer::Target;
 
+/// The data referenced by a `Buffer`.
+pub enum Data<'a> {
```
Comment (alteous) [inline src/buffer.rs]: Closes #106

**Comment by alteous** [comment]
This is a long way off.

**Pull request closed without merging**